pages/profile/index.js

```diff
@@ -0,0 +1,313 @@
+    if (isEditing) {
+      setIsEditing(false);
+    } else {
+      router.push("/");
+    }
+  }
+
+  const title = isEditing
+    ? profile.firstName || profile.lastName
+      ? "Edit Profile"
+      : "Create Profile"
+    : "Profile";
+
+  return (
+    <>
+      {" "}
+      <StyledBackLink as="button" onClick={handleBack}>
+        <StyledImage
+          aria-hidden="true"
+          src="/images/arrow-return-left.svg"
+          alt="edit button"
+          width={15}
+          height={15}
+        />
+        Back
+      </StyledBackLink>
+      <StyledProfileContainer>
+        <StyledTitle>{title}</StyledTitle>
+        <EditButton onClick={handleEdit}>
+          <StyledImage
+            aria-hidden="true"
+            src="/images/pencil.svg"
+            alt="edit button"
+            width={15}
+            height={15}
+          />
+        </EditButton>
+      </StyledProfileContainer>
+      <Container>
+        {!isEditing ? (
+          <ProfileView>
+            <Avatar src={profile.avatar} alt="Profile Avatar" />
+            <Name>
+              {`${profile.firstName} ${profile.lastName}` || "Your Name"}
+            </Name>
+          </ProfileView>
+        ) : (
+          <ProfileEdit>
+            <AvatarSelection>
+              <p>Select Avatar:</p>
+              {avatars.map((avatar) => (
+                <AvatarOption
+                  key={avatar}
+                  isSelected={tempProfile.avatar === avatar}
+                  onClick={() => handleAvatarChange(avatar)}
+                >
+                  <AvatarSelectionImage src={avatar} alt="Avatar Option" />
+                </AvatarOption>
+              ))}
+            </AvatarSelection>
+            <InputField>
+              <label>First Name</label>
+              <input
+                type="text"
+                maxLength={20}
+                value={tempProfile.firstName}
+                onChange={(event) =>
+                  setTempProfile((previous) => ({
+                    ...previous,
+                    firstName: event.target.value,
+                  }))
+                }
+              />
+              <Counter>{tempProfile.firstName.length}/20</Counter>
+            </InputField>
+            <InputField>
+              <label>Last Name</label>
+              <input
+                type="text"
+                maxLength={20}
+                value={tempProfile.lastName}
+                onChange={(event) =>
+                  setTempProfile((previous) => ({
+                    ...previous,
+                    lastName: event.target.value,
+                  }))
+                }
+              />
+              <Counter>{tempProfile.lastName.length}/20</Counter>
+            </InputField>
+            <ButtonRow>
+              <CancelButton onClick={handleCancel}>Cancel</CancelButton>
+              <SaveButton onClick={handleSave}>Save</SaveButton>
+            </ButtonRow>
+          </ProfileEdit>
```

Comment:
This would be the perfect situation to use a `form` element. Instead of using a `tempProfile` state, you can use the data you receive upon submitting with `FormData`. Further you could assign `defaultValue`s instead of the value + onChange input handling that you are doing right now. When the user clicks "save", the data is collected in the handleSubmit function and the `profile` state is set to the respective values you got from FormData. If you press cancel, the form is resetted and you display the information stored in the `profile` state. 

This approach has these benefits: 
- one state less (no need for `tempProfile`)
- usage of HTML features -> less rerenders of the component
- no onChange handling required -> code is simpler
- less code overall
- consistency with other parts of the app, where you already use FormData and form elements.

Reply:
Thank you for your feedback.
I´ve refactored the code, and it is for sure less code and simpler.